MdeModulePkg/Core/Dxe/Misc/MemoryProtection.c

```diff
@@ -863,14 +1106,70 @@ MemoryProtectionExitBootServicesCallback (
   // delay setting protections on RT code pages until after SetVirtualAddressMap().
   // OS may set protection on RT based upon EFI_MEMORY_ATTRIBUTES_TABLE later.
   //
-  if (mImageProtectionPolicy != 0) {
+  // MU_CHANGE START Update to use memory protection settings HOB
+  // if (mImageProtectionPolicy != 0) {
+  if (gDxeMps.ImageProtectionPolicy.Data) {
+    // MU_CHANGE END
     for (Link = gRuntime->ImageHead.ForwardLink; Link != &gRuntime->ImageHead; Link = Link->ForwardLink) {
       RuntimeImage = BASE_CR (Link, EFI_RUNTIME_IMAGE_ENTRY, Link);
       SetUefiImageMemoryAttributes ((UINT64)(UINTN)RuntimeImage->ImageBase, ALIGN_VALUE (RuntimeImage->ImageSize, EFI_PAGE_SIZE), 0);
     }
   }
 }
 
+// MU_CHANGE [START]: Move disable NULL detection to separate routine so it can be called
+//                    outside of the event context.
+
+/**
+  Disable NULL pointer detection.
+**/
+VOID
+DisableNullDetection (
+  VOID
+  )
+{
+  EFI_STATUS                       Status;
+  EFI_GCD_MEMORY_SPACE_DESCRIPTOR  Desc;
+
+  DEBUG ((DEBUG_INFO, "%a - start\r\n", __func__));

```

Comment:
Is there a particular reason for the carriage return here?


```suggestion
  DEBUG ((DEBUG_INFO, "%a - start\n", __func__));
```

Reply:
If it is removed here, it should be removed from the comment at the end of the function as well.

Reply:
Looks like it was in the original code, but I still think it's unnecessary.